File: src/checkoutservice/examples/placeorder.js
/**
 * Using: https://github.com/njpatel/grpcc
 * src/checkoutservice $ grpcc --insecure -p proto/demo.proto -a localhost:5050 -s hipstershop.CheckoutService --exec examples/placeorder.js
 */
client.placeOrder(
  {
    user_id: '123',
    user_currency: 'USD',
    address: {
      street_address: '123 foo bar lane',
      city: 'san francisco',
      state: 'CA',
      country: 'USA',
      zip_code: 94329,
    },
    email: 'foo@bar.com',
    credit_card: {
      credit_card_number: '4009366231016609',
      credit_card_cvv: 123,
      credit_card_expiration_year: 2025,
      credit_card_expiration_month: 12,
    },
  },
  printReply
);

File: src/clients/grpc/golang/cmd/order/placeOrder.go
package main

import (
	"context"
	"fmt"

	pb "gcp/demo"

	"google.golang.org/grpc"
)

func main() {
	conn, err := grpc.Dial("checkoutservice:5050", grpc.WithInsecure())
	if err != nil {
		fmt.Printf("Could not connect: %v\n", err)
	}

	defer conn.Close()
	c := pb.NewCheckoutServiceClient(conn)

	r, err := c.PlaceOrder(context.Background(), &pb.PlaceOrderRequest{
		UserId:       "abcde",
		UserCurrency: "USD",
		Address: &pb.Address{
			StreetAddress: "1600 Amp street",
			City:          "Mountain View",
			State:         "CA",
			Country:       "USA",
			ZipCode:       94043,
		},
		Email: "someone@example.com",
		CreditCard: &pb.CreditCardInfo{
			CreditCardNumber:          "4432-8015-6152-0454",
			CreditCardCvv:             672,
			CreditCardExpirationYear:  2024,
			CreditCardExpirationMonth: 1,
		},
	})
	if err != nil {
		fmt.Printf("Something went wrong: %v\n", err)
		return
	}

	fmt.Println("Order result: ", r.Order)
}
